packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraPlugin.java

```diff
@@ -45,599 +39,679 @@
-          for (String cameraName : cameraNames) {
-            HashMap<String, Object> details = new HashMap<>();
-            CameraCharacteristics characteristics =
-                cameraManager.getCameraCharacteristics(cameraName);
-            details.put("name", cameraName);
-            @SuppressWarnings("ConstantConditions")
-            int lens_facing = characteristics.get(CameraCharacteristics.LENS_FACING);
-            switch (lens_facing) {
-              case CameraMetadata.LENS_FACING_FRONT:
-                details.put("lensFacing", "front");
+                camera = new Camera(cameraName, resolutionPreset, result);
                 break;
-              case CameraMetadata.LENS_FACING_BACK:
-                details.put("lensFacing", "back");
+            }
+            case "takePicture": {
+                camera.takePicture((String) call.argument("path"), result);
                 break;
-              case CameraMetadata.LENS_FACING_EXTERNAL:
-                details.put("lensFacing", "external");
+            }
+            case "startVideoRecording": {
+                final String filePath = call.argument("filePath");
+                camera.startVideoRecording(filePath, result);
                 break;
             }
-            cameras.add(details);
-          }
-          result.success(cameras);
-        } catch (CameraAccessException e) {
-          result.error("cameraAccess", e.getMessage(), null);
-        }
-        break;
-      case "create":
-        {
-          FlutterView.SurfaceTextureEntry surfaceTexture = view.createSurfaceTexture();
-          final EventChannel eventChannel =
-              new EventChannel(
-                  registrar.messenger(),
-                  "flutter.io/cameraPlugin/cameraEvents" + surfaceTexture.id());
-          String cameraName = call.argument("cameraName");
-          String resolutionPreset = call.argument("resolutionPreset");
-          Cam cam = new Cam(eventChannel, surfaceTexture, cameraName, resolutionPreset, result);
-          cams.put(cam.getTextureId(), cam);
-          break;
-        }
-      case "start":
-        {
-          Cam cam = getCamOfCall(call);
-          cam.start();
-          result.success(null);
-          break;
-        }
-      case "capture":
-        {
-          Cam cam = getCamOfCall(call);
-          cam.capture((String) call.argument("path"), result);
-          break;
-        }
-      case "stop":
-        {
-          Cam cam = getCamOfCall(call);
-          cam.stop();
-          result.success(null);
-          break;
-        }
-      case "dispose":
-        {
-          Cam cam = getCamOfCall(call);
-          if (cam != null) {
-            cam.dispose();
-          }
-          cams.remove(textureIdOfCall(call));
-          result.success(null);
-          break;
+            case "stopVideoRecording": {
+                camera.stopVideoRecording(result);
+                break;
+            }
+            case "start": {
```

Comment:
@nisdis what is the difference between startCamera/closeCamera start/stop? do we need start/stop?

Reply:
@luigi-agosti 
This was made as a dummy switch statement as not to throw any errors on Android.
However the iOS code didn't work because the preview wasn't started with the `start` call.
I need to figure a workaround to remove this on the iOS side too.
Its not required on the android side as @quentinleguennec has removed the unnecessary feature of the original plugin.